test/e2e/provider_expectations.go

```diff
@@ -198,3 +199,119 @@ MachineLoop:
+					machine = machineItem
+					nodeName = machineItem.Status.NodeRef.Name
+					return true, nil
+				}
+			}
+		}
+		return false, fmt.Errorf("No machine found with NodeRef not set")
+	}); err != nil {
+		return err
+	}
+
+	glog.Info("Get nodeList")
+	if err := wait.PollImmediate(1*time.Second, waitShort, func() (bool, error) {
+		nodeList := corev1.NodeList{}
+		if err := tc.client.List(context.TODO(), &listOptions, &nodeList); err != nil {
+			glog.Errorf("error querying api for nodeList object: %v, retrying...", err)
+			return false, nil
+		}
+		for _, nodeItem := range nodeList.Items {
+			if nodeItem.Name == nodeName {
+				node = &nodeItem
+				break
+			}
+		}
+		if node == nil {
+			return false, fmt.Errorf("Node %q not found", nodeName)
+		}
+		return true, nil
+	}); err != nil {
+		return err
+	}
+
+	glog.Info("Annotate machine with `drain-node` annotation")
+	if machine.ObjectMeta.Annotations == nil {
+		machine.ObjectMeta.Annotations = make(map[string]string)
+	}
+	machine.ObjectMeta.Annotations["openshift.io/drain-node"] = "True"
+	if err := tc.client.Update(context.TODO(), &machine); err != nil {
+		return fmt.Errorf("Unable to set `drain-node` annotation")
+	}
+
+	currentTimestamp := metav1.Now()
+
+	glog.Info("Delete machine and observe node draining")
+	if err := tc.client.Delete(context.TODO(), &machine); err != nil {
+		return fmt.Errorf("Unable to delete %q machine", machine.Name)
+	}
+
+	return wait.PollImmediate(time.Second, waitShort, func() (bool, error) {
+		eventList := corev1.EventList{}
+		if err := tc.client.List(context.TODO(), &listOptions, &eventList); err != nil {
+			glog.Errorf("error querying api for eventList object: %v, retrying...", err)
+			return false, nil
+		}
+
+		glog.Infof("Fetching delete machine and node drained events")
+		var nodeDrainedEvent *corev1.Event
+		var machineDeletedEvent *corev1.Event
+		for _, eventItem := range eventList.Items {
+			// Skip all events recorded before the Delete operation
+			if eventItem.FirstTimestamp.Before(&currentTimestamp) {
+				continue
+			}
+
+			if eventItem.Reason == "Deleted" && eventItem.Message == fmt.Sprintf("Node %q drained", nodeName) {
+				nodeDrainedEvent = &eventItem
+				continue
+			}
+			// always take the newest 'machine deleted' event
+			if eventItem.Reason == "Deleted" && eventItem.Message == fmt.Sprintf("Deleted machine %v", machine.Name) {
+				machineDeletedEvent = &eventItem
+			}
+		}
+
+		if nodeDrainedEvent == nil {
+			glog.Infof("Unable to find %q node drained event", nodeName)
+			return false, nil
+		}
+
+		if machineDeletedEvent == nil {
+			glog.Infof("Unable to find %q machine deleted event", machine.Name)
+			return false, nil
+		}
+
+		glog.Infof("Node %q drained event recorded: %#v", nodeName, *nodeDrainedEvent)
+
+		if machineDeletedEvent.FirstTimestamp.Before(&nodeDrainedEvent.FirstTimestamp) {
+			err := fmt.Errorf("Machine %q deleted before node %q got drained", machine.Name, nodeName)
+			glog.Info(err)
```

Comment:
why not just `glog.errorf`?

Reply:
Ha, yes, it's suppose to be `glog.Errorf`